source/NoteWindow.cpp

```diff
@@ -77,111 +78,66 @@ const struct tm gettime() {
-	CreateOtherWindows();
-
-	// The main view is created
-	frameView = Bounds();
-	frameView.top = fNoteMenuBar->Bounds().Height() + 1;
-
-	fNoteView = new NoteView (frameView, B_FOLLOW_ALL, false, this);
-
-	// Text and Scroll View
-	frameView = fNoteView -> Bounds();
-	frameView.top += 10;
-	frameView.right -= B_V_SCROLL_BAR_WIDTH;
-	frameView.bottom -= B_H_SCROLL_BAR_HEIGHT;
-	frameView.left = 0;
-	frameText = frameView;
-
-	frameText.OffsetTo(B_ORIGIN);
-	frameText.InsetBy(TEXT_INSET, TEXT_INSET);
-
-	fNoteText = new NoteText(frameView, frameText, "NoteText", this);
-	fNoteText->SetDoesUndo(true);
-	fNoteText->MakeFocus();
-	fNoteText->SetStylable(true);
-
-	// ScrollView
-	fScrollView = new BScrollView("scrollview", fNoteText, B_FOLLOW_ALL, 0, true, true, B_NO_BORDER);
+	_CreateNoteView();
 
 	// Set the title
 	title << id;
 	SetTitle(title.String());
 
 	// It will be associated to the window
 	AddChild(fNoteView);
-	fNoteView -> AddChild(fScrollView);
 
 	// Creating the file panel
-	fSavePanel = new BFilePanel (B_SAVE_PANEL, new BMessenger (this), directory, B_FILE_NODE, false, NULL,
-				refFilter, false, true);
+	fSavePanel = new BFilePanel(B_SAVE_PANEL, new BMessenger(this), NULL, 0,
+		false);
 
 	Show();
 }
 
 // Construtor
 NoteWindow :: NoteWindow(entry_ref *ref)
 	: BWindow (BRect(100,100,350,350) , "TakeNotes", B_DOCUMENT_WINDOW, B_ASYNCHRONOUS_CONTROLS){
-	SetSizeLimits(250, 10000, 250, 10000);
 
 	// Variables
 	BMessage 	*msg = new BMessage;
 	BFile		f;
 	status_t 	result;
-	entry_ref	*directory = NULL;
-	BRefFilter	*refFilter = NULL;
 	BAlert 		*alert;
-	BEntry 		entry(ref);
+	BEntry		entry(ref, true);	// entry of possibly linked file
 	BRect		viewRect;
-	char 		name[B_FILE_NAME_LENGTH];
-
-	// Initialize the messenger: the handler is the window itself
-	fMessenger = BMessenger(this);
+	char 		name[B_FILE_NAME_LENGTH] = "Untitled Note 0";
+	bool		isNoteFile = false;	// readable takenotes file
+	bool		isNewFile = false;
 
 	// Initialize all the "static" elements like MenuBar,Menu,MenuItems and so on
 	InitWindow();
 
-	// Create the Alarm, Choice, Background and Tags Window
-	CreateOtherWindows();
-
-	// Check if the file exists and if it is loadable
-	result = f.SetTo(ref, B_READ_ONLY);
-		switch(result){
-
-			case B_OK:
-				printf("ok\n");
-			break;
-
-			case B_BAD_VALUE:
-				printf("path nullo\n");
-			break;
-
-			case B_ENTRY_NOT_FOUND:{
-				printf("file not found\n");
-				alert = new BAlert("File Not Found","file not found", "OK");
-				alert->Go();
+	// Check if the file exists and if it is a loadable takenotes file
+	if (entry.InitCheck() == B_OK)
+		if (entry.Exists()) {
+			BNode node(&entry);
+			if (node.InitCheck() == B_OK) {
+				BNodeInfo nodeInfo(&node);
+				if (nodeInfo.InitCheck() == B_OK) {
+					char type[B_MIME_TYPE_LENGTH];
+					if (nodeInfo.GetType(type) == B_OK &&
+						strcmp(type, "application/takenotes") == 0 &&
+						f.SetTo(ref, B_READ_ONLY) == B_OK)
+						isNoteFile = true;
+				}
```

Comment:
You can reduce one level of indent by moving the declaration of `type` to the parent scope and then combining the two if-statements.

Reply:
Thanks for the trick!